internal/gps/source.go

```diff
@@ -548,7 +552,7 @@ type source interface {
 	updateLocal(context.Context) error
 	listVersions(context.Context) ([]PairedVersion, error)
 	getManifestAndLock(context.Context, ProjectRoot, Revision, ProjectAnalyzer) (Manifest, Lock, error)
-	listPackages(context.Context, ProjectRoot, Revision) (pkgtree.PackageTree, error)
+	listPackages(context.Context, string, ProjectRoot, Revision) (pkgtree.PackageTree, error)
```

Comment:
> this could just be a method on PackageTree itself;

I actually had that and hit some issues due to the cache. Didn't dig any deeper and just hacked `listPackages`. You have a better overview for a more elegant implementation.